platforms/jvm/plugins-java/src/main/java/org/gradle/jvm/component/internal/DefaultJvmSoftwareComponent.java

```diff
@@ -184,4 +198,66 @@ public void withSourcesJar() {
     public JvmFeatureInternal getMainFeature() {
         return mainFeature;
     }
+
+    @Nullable
+    public JvmTestSuite getTestSuite() {
+        return testSuite;
+    }
+
+    @Override
+    public void useCompileClasspathConsistency() {
+        if (testSuite != null) {
+            configurations.getByName(testSuite.getSources().getCompileClasspathConfigurationName())
+                .shouldResolveConsistentlyWith(mainFeature.getCompileClasspathConfiguration());
+        }
+    }
+
+    @Override
+    public void useRuntimeClasspathConsistency() {
+        if (testSuite != null) {
+            configurations.getByName(testSuite.getSources().getRuntimeClasspathConfigurationName())
+                .shouldResolveConsistentlyWith(mainFeature.getRuntimeClasspathConfiguration());
+        }
+    }
+
+    private JvmTestSuite configureBuiltInTest(Project project) {
+        /*
+         * At some point we may want to create a test suite per component, but for now we only want to create one for the `java`
+         * component, in case multiple DefaultJvmSoftwareComponents are created.
+         */
+        TestingExtension testing = project.getExtensions().findByType(TestingExtension.class);
+        if (null != testing && JvmConstants.JAVA_COMPONENT_NAME.equals(getName())) {
+            final NamedDomainObjectProvider<JvmTestSuite> testSuite = testing.getSuites().register(JavaPluginHelper.DEFAULT_TEST_SUITE_NAME, JvmTestSuite.class, suite -> {
```

Comment:
I still think we want to move this back to where it was before these changes -- we just don't need to do all the _other_ enhancements to this class in this PR. 

This means we should restore the `configureBuiltInTest` method in `JavaPlugin` and replace the old `configureBuiltInTest(project, component)` line with something like 

```
Provider<JvmTestSuite> defaultTestSuite = configureBuiltInTest(project, component)
component.getTestSuites().addLater(defaulTestSuite)
```

That would also mean the `DefaultJvmSoftwareComponent` should have a list of test suites and the `useRuntimeClasspathConsistency` and `useCompileClasspathConsistency` methods would apply consistency to all of its test suites (even if there is only one)


However, I am also fine if we delay this one, as this change should not affect behavior.  If you'd rather get these existing changes pushed through, its probably fine to make this one change in a small follow-up PR